File: authentication-server/src/main/java/com/andrew/authenticationserver/repository/UserRepository.java
package com.andrew.authenticationserver.repository;

import com.andrew.authenticationserver.entity.auth.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.jpa.repository.Query;
import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
import org.springframework.data.repository.query.Param;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    @Query("select u from User u " +
            "where u.email = :email and u.status ='A'")
    Optional<User> findByEmail(@Param("email") String email);

    @Query("select u from User u " +
            "where u.email = :email and u.status ='W'")
    Optional<User> findWaitingByEmail(@Param("email") String email);
}

File: auth-server/src/main/java/com/andrew/authserver/repository/UserRepository.kt
package com.andrew.authserver.repository

import com.andrew.authserver.entity.auth.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository : JpaRepository<User, Long>{
    fun findByEmail(email: String): Optional<User>;
    fun findByUsername(username: String);
}